static/app/components/sidebar/newOnboardingStatus.tsx

```diff
@@ -0,0 +1,192 @@
+import {Fragment, useCallback, useContext, useEffect} from 'react';
+import type {Theme} from '@emotion/react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {OnboardingContext} from 'sentry/components/onboarding/onboardingContext';
+import {
+  NewOnboardingSidebar,
+  useOnboardingTasks,
+} from 'sentry/components/onboardingWizard/newSidebar';
+import ProgressRing, {
+  RingBackground,
+  RingBar,
+  RingText,
+} from 'sentry/components/progressRing';
+import {ExpandedContext} from 'sentry/components/sidebar/expandedContextProvider';
+import {t, tct} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {isDemoWalkthrough} from 'sentry/utils/demoMode';
+import theme from 'sentry/utils/theme';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+
+import type {CommonSidebarProps} from './types';
+import {SidebarPanelKey} from './types';
+
+type NewOnboardingStatusProps = CommonSidebarProps;
+
+export function NewOnboardingStatus({
+  collapsed,
+  currentPanel,
+  orientation,
+  hidePanel,
+  onShowPanel,
+}: NewOnboardingStatusProps) {
+  const organization = useOrganization();
+  const onboardingContext = useContext(OnboardingContext);
+  const {projects} = useProjects();
+  const {shouldAccordionFloat} = useContext(ExpandedContext);
+
+  const isActive = currentPanel === SidebarPanelKey.ONBOARDING_WIZARD;
+  const walkthrough = isDemoWalkthrough();
+
+  const {allTasks, gettingStartedTasks, beyondBasicsTasks, completeTasks} =
+    useOnboardingTasks(organization, projects, onboardingContext);
+
+  const handleToggle = useCallback(() => {
+    if (!walkthrough && !isActive === true) {
+      trackAnalytics('quick_start.opened', {
+        organization,
+      });
+    }
+    onShowPanel();
+  }, [walkthrough, isActive, onShowPanel, organization]);
+
+  const label = walkthrough ? t('Guided Tours') : t('Onboarding');
+  const totalRemainingTasks = allTasks.length - completeTasks.length;
+  const pendingCompletionSeen = completeTasks.some(
+    completeTask =>
+      allTasks.some(task => task.task === completeTask.task) &&
+      completeTask.status === 'complete' &&
+      !completeTask.completionSeen
+  );
```

Comment:
this logic differs slightly from what was previously implemented in the `OnboardingStatus` component. For instance, we now display all tasks, ensuring that the counter decreases consistently as users would expect.